src/app/LoginPage/LoginPage.tsx

```diff
@@ -89,22 +128,94 @@ class LoginPage extends React.Component<LoginProps, LoginState> {
+      });
+  }
+
+  resetPassword = (event: React.MouseEvent) => {
+    event.preventDefault();
+    let apiUrl = "https://cce4c6f9-ed0c-4bd9-94c7-6f174f7ae024.mock.pstmn.io/reset"
+    const credential = {
+      oldPassword: this.state.oldPassword,
+      newPassword: this.state.newPassword
+    };
+    this.setState({ isValidOldPassword: undefined })
+    axios
+      .post(apiUrl, { credential })
+      .then(res => {
+        debugger
+        if (res && res.data) {
+          if (res.data["status"] == "true") {
+            this.setState({ ...this.state, isResetPasswordProcessed: true });
+          }
+          else {
+            this.setState({ ...this.state, isResetPasswordProcessed: false, error: "Invalid Username/Password" });
+          }
+          this.setState({ ...this.state, oldPassword: "", newPassword: "", confirmPassword: "" });
+        }
+      })
+      .catch(err => {
+        this.setState({ ...this.state, isResetPasswordProcessed: false, error: err });
+      });
+  }
+
   handleSubmit = (event: React.MouseEvent) => {
     event.preventDefault();
+    this.setState({ error: "" })
     // dynamic call
     const { username, password } = this.state;
     let role: Role;
-    if (username === 'admin' && password === 'adminpass') {
-      role = Role.ADMIN;
-      localStorage.setItem('login', 'ADMIN');
-    } else if ((username === 'developer1' && password === 'developer1pass') || (username === 'developer2' && password === 'developer2pass')) {
-      role = Role.DEVELOPER;
-      localStorage.setItem('login', 'DEVELOPER');
-    } else {
-      role = Role.NONE;
-      this.setState({ ...this.state, error: "Invalid Username/Password" })
-    }
-    this.props.setRole(role);
+    let apiUrl = this.state.api;
+    const credential = {
+      username: username,
+      password: password
+    };
+    debugger
+    axios
+      .post(apiUrl, { credential })
+      .then(res => {
+        if (res && res.data) {
+          if (res.data["is_valid"] == "true") {
+            this.setState({ ...this.state, isValidUser: true });
+            role = Role.ADMIN;
+            this.props.setRole(role);
+          }
+          else {
+            this.setState({ ...this.state, isValidUser: false, error: "Invalid Username/Password" });
+          }
+        }
+      })
+      .catch(err => {
+        this.setState({ ...this.state, isValidUser: false, error: err });
+      });
+
+    // if (username === 'admin' && password === 'adminpass') {
+    //   role = Role.ADMIN;
+    //   localStorage.setItem('login', 'ADMIN');
+    // } else if ((username === 'developer1' && password === 'developer1pass') || (username === 'developer2' && password === 'developer2pass')) {
+    //   role = Role.DEVELOPER;
+    //   localStorage.setItem('login', 'DEVELOPER');
+    // } else {
+    //   role = Role.NONE;
+    //   this.setState({ ...this.state, error: "Invalid Username/Password" })
+    // }
+    // this.props.setRole(role);
```

Comment:
You aren't checking for the password match and hence, the application is taking every input as a valid password. Please review this.

Reply:
We will check password after retrieving values from the database.